src/plugins/intel_cpu/src/transformations/transformation_pipeline.cpp

```diff
@@ -565,32 +571,47 @@ void Transformations::Lpt(const bool hasINT16orINT32Levels, const std::vector<ov
     }
 
     ov::pass::Manager lptManager;
-    CPU_REGISTER_PASS_COMMON(lptManager, ov::pass::low_precision::LowPrecision,
+    CPU_REGISTER_PASS_COMMON(lptManager, LowPrecision,
         supportedPrecisions,
         quantizationRestrictions,
         LayerTransformation::Params(updatePrecision, ov::element::f32, defaultPrecisions));
-    CPU_SET_CALLBACK_COMMON(lptManager,
-        [](const_node_ptr& node) -> bool {
-            if (const auto mulitply = std::dynamic_pointer_cast<const ov::opset1::Multiply>(node)) {
-                return !MultiplyToGroupConvolutionTransformation::canBeTransformedToGroupConvolution(mulitply);
-            }
-            return false;
-        },
-        ov::pass::low_precision::MarkupPrecisions);
-    CPU_SET_CALLBACK_COMMON(lptManager,
-        [&defaultPrecisions](const_node_ptr& node) -> bool {
-            return LayerTransformation::isAsymmetricQuantization(node, defaultPrecisions) ||
-                WeightableLayerTransformation::isAsymmetricOnWeights(node, defaultPrecisions);
-        },
-        ov::pass::low_precision::ConvolutionBackpropDataTransformation);
 
-    lptManager.get_pass_config()->set_callback<ov::pass::low_precision::AddTransformation>(
-        [](const_node_ptr& node) -> bool {
-            return ov::marked_as_bias(node);
-        });
+    CPU_SET_CALLBACK_COMMON(lptManager, [](const_node_ptr& node) -> bool {
+        return ov::is_type<ov::opset1::Multiply>(node) &&
+               !MultiplyToGroupConvolutionTransformation::canBeTransformedToGroupConvolution(node);
+    }, MarkupPrecisions);
+    CPU_SET_CALLBACK_COMMON(lptManager, [&defaultPrecisions](const_node_ptr& node) -> bool {
+        return LayerTransformation::isAsymmetricQuantization(node, defaultPrecisions) ||
+               WeightableLayerTransformation::isAsymmetricOnWeights(node, defaultPrecisions);
+    }, ConvolutionBackpropDataTransformation);
+    CPU_SET_CALLBACK_COMMON(lptManager, [](const_node_ptr& node) -> bool {
+        return ov::marked_as_bias(node);
+    }, AddTransformation);
+
+    CPU_SET_CALLBACK_X64(lptManager, [&](const_node_ptr& node) -> bool {
+        const auto& consumers = node->get_output_target_inputs(0);
+        if (consumers.size() == 1) {
+            const auto consumer = consumers.begin()->get_node()->shared_from_this();
+            return ov::is_type<ov::opset1::Multiply>(consumer) && is_decompression_multiply(consumer);
+        }
+        return false;
+    }, FoldConvertTransformation);
```

Comment:
Isn't it better to change FoldConvertTransformation behavior in order to respect DisableConstantFolding attr?

Reply:
`FoldConvertTransformation` was designed especially to fold Convert ignoring the `DisableConstantFolding` attribute.
Historically, the attribute was used to keep dequantization subgraphs on weights as is till the LPT stage. If LPT fail to propagate some of the subgraphs, `FoldConvertTransformation` helps to fold them in constant.

Reply:
I am ok to keep the solution for now, but would insist on follow-up in future.
Seems like better solution is to limit applicability of this transformation. Ok, we want to fold Convert operations even DisableConstantFolding is set to true, but lets do it only for quantized Matmuls, not all of them. 
BTW do we have examples why we need to fold Convert ops on weights inside LPT? Some sort of canonicalization? Maybe we can disable this transformation at all for CPU?

Reply:
> Ok, we want to fold Convert operations even DisableConstantFolding is set to true, but lets do it only for quantized Matmuls, not all of them.

This transformation was needed for all types of Convolutions too, not only for Matmuls.
> BTW do we have examples why we need to fold Convert ops on weights inside LPT? Some sort of canonicalization?

Yes, we need it to guarantee that after LPT there will be no dequantization subgraphs on weights. This allows us not to take into account the cases with dequantization on weighs in further transformations.
> Maybe we can disable this transformation at all for CPU?

At first glance, I see 4 consequences of the disabling (only in case when it is not possible to propagate dequantization through the corresponding layer):

1. Dequantization subgraphs on weights will be tokenized in Snippets for all layers except Matmul (for Matmul we have a decompression handling related checks which will work for dequantization subgraphs too).
2.  ConvertMatMulToFC will be called for the MatMuls with not folded weights - it is not a problem since the transformation has been adapted to such subgraphs as part of decompressed weights support.
3. Dequantization subgraph might be fused into FullyConnected as decompression on x64 platforms.
4. The rest Dequantization subgraphs will be folded by the plugin.

If you are OK with these consequences, I can create a ticket in which we will disable this transformation and conduct experiments on a models scope.

Reply:
All that you are describing highlights that LPT needs to do weights canonicalization, but only for operations which are quantized. All others (decompressed or pure floating point) should be untouched. 
Lets create a ticket for that. I understand some pipeline adjustments will be needed, but from deisgn standpoint that seems to be correct way forward.

Reply:
The ticket is created: 125023